diff --git a/lib/layout/grid.js b/lib/layout/grid.js
index 2bb5d43ec6784176cbd0a389824bf4ff1e8c44b6..186cf60812e1b3fece082ca1de331245d8cc270a 100755
--- a/lib/layout/grid.js
+++ b/lib/layout/grid.js
@@ -7,6 +7,8 @@ function Grid(options) {
   if (!options.screen) throw 'Error: A screen property must be specified in the grid options.\r\n' +
                               'Note: Release 2.0.0 has breaking changes. Please refer to the README or to https://github.com/yaronn/blessed-contrib/issues/39';
   this.options = options;
+  this.options.top = this.options.top || 0;
+  this.options.left = this.options.left || 0;
   this.options.dashboardMargin = this.options.dashboardMargin || 0;
   this.cellWidth = ((100 - this.options.dashboardMargin*2) / this.options.cols);
   this.cellHeight = ((100  - this.options.dashboardMargin*2) / this.options.rows);
@@ -25,8 +27,8 @@ Grid.prototype.set = function(row, col, rowSpan, colSpan, obj, opts) {
   //var options = JSON.parse(JSON.stringify(opts));
   var options = {};
   options = utils.MergeRecursive(options, opts);
-  options.top = top + '%';
-  options.left = left + '%';
+  options.top = top + '%+' + this.options.top;
+  options.left = left + '%+' + this.options.left;
   options.width = (this.cellWidth * colSpan - widgetSpacing) + '%';
   options.height = (this.cellHeight * rowSpan - widgetSpacing) + '%';
   if (!this.options.hideBorder)
